src/txn/txn.c

```diff
@@ -440,11 +440,37 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_RET(__wt_config_gets_def(session, cfg, "read_timestamp", 0, &cval));
 	if (cval.len > 0) {
 #ifdef HAVE_TIMESTAMPS
+		bool round_to_oldest;
+		int cmp;
 		wt_timestamp_t ts;
+		WT_TXN_GLOBAL *txn_global;
 
+		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-		WT_RET(__wt_timestamp_validate(session,
-		    &ts, &cval, true, false, false));
+		WT_WITH_TIMESTAMP_READLOCK(session, &txn_global->rwlock,
+		    cmp = __wt_timestamp_cmp(&ts,
+		    &txn_global->oldest_timestamp));
```

Comment:
You should check if txn_global->has_oldest_timestamp is set before comparing with txn_global->oldest_timestamp. For example like what you have in __wt_timestamp_validate, you can do:

```
WT_WITH_TIMESTAMP_READLOCK(session, &txn_global->rwlock,
    older_than_oldest_ts = (txn_global->has_oldest_timestamp &&
        __wt_timestamp_cmp(&ts, &txn_global->oldest_timestamp) < 0));
if (older_than_oldest_ts) {
    // blah blah
}
```